### PR TITLE
add do better web audit references

### PR DESCRIPTION
**Pull request by kaycebasques**
ok @ebidel you're up, plz review

### CODE REVIEW

**Comment by jpmedley** [comment]
An absolute path in one of the files is breaking this.

`ERROR: src/content/en/tools/lighthouse/audits/noopener.md Hard coded https://developers.google.com in link (HTML) -- href="https://developers.google.com`

FYI, you can do this same check locally with `gulp test`.

File: src/content/en/tools/lighthouse/_toc.yaml [hunk 1]
```diff
@@ -39,10 +39,32 @@ toc:
         path: /web/tools/lighthouse/audits/manifest-short_name-is-not-truncated
       - title: Page Contains Some Content When Its Scripts Are Not Available
         path: /web/tools/lighthouse/audits/no-js
+      - title: Page Does Not Automatically Request Geolocation On Page Load
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/_toc.yaml]: Notification Permissions -> Notification Permission

Comment (ebidel) [inline src/content/en/tools/lighthouse/_toc.yaml]: Geolocation -> Geolocation Permission

File: src/content/en/tools/lighthouse/audits/blocking-links.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+You can improve your page load speed by inlining resources that are required
+for first paint, and deferring those that aren't.
+
+## How to pass the audit {: #how }
+
+In your report, Lighthouse lists all of the render-blocking links that it has
+detected. The goal is to reduce this number.
+
+As mentioned in [How the audit is implemented](#implementation), Lighthouse
+flags three types of render-blocking links: scripts, stylesheets, and HTML
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: The title of this page is just about links, but the info covers both the link and script audit. I think having a single page for both rendering blocking audits is fine, but we should update the title to reflect the content.

File: src/content/en/tools/lighthouse/audits/blocking-links.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+imports. How you optimize depends on what type of link you're working with.
+
+Note: When a resource is referred to as "critical" below, it means that the
+resource is required for first paint.
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: It's more than just FP. "required for first paint or crucial to the app's functionality."

File: src/content/en/tools/lighthouse/audits/blocking-links.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+* For critical scripts, consider inlining them in your HTML. For non-critical
+  scripts, consider marking them with the `async` or `defer` attributes.
+  See [Adding Interactivity with JavaScript][js] to learn more.
+* For stylesheets, consider splitting up your styles into different files,
+  organized by media query, and then adding a `media` attribute to each
+  stylesheet link. When loading a page, the browser only blocks the first
+  paint to retrieve the stylesheets that match the user's device. See
+  [Render-Blocking CSS][css] to learn more.
+* For non-critical HTML imports, mark them with the `async` attribute.
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: For imports, `async` should be used all the time if the user can get away with it.

File: src/content/en/tools/lighthouse/audits/blocking-links.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+* Is in the `<head>` of the document.
+* Does not have a `defer` attribute.
+* Does not have an `async` attribute.
+
+A `<link rel="stylesheet">` tag that:
+
+* Does not have a `disabled` attribute. When this attribute is present,
+  the browser does not download the stylesheet.
+* Does not have a `media` attribute.
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: ...attribute that matches the user's device.

File: src/content/en/tools/lighthouse/audits/console-time.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+## Why the audit is important {: #why }
+
+If you're using `console.time()` to measure your page's performance, consider
+using the User Timing API instead. Benefits include:
+
+* High-resolution timestamps.
+* Exportable timing data.
+* Integration with the Chrome DevTools Timeline. When the User Timing function
+  `performance.measure()` is called during a Timeline recording, DevTools
+  automatically adds the measurement to the Timeline's results.
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/console-time.md]: would be cool to include a screenshot of the timeline annotated with a user timing. chromestatus.com does some user timings if you want to take it from there.

File: src/content/en/tools/lighthouse/audits/geolocation-on-load.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+See [Ask permission responsibly][ask] for a list of best practices when
+requesting a user's location.
+
+[ask]: /web/fundamentals/native-hardware/user-location/#ask_permission_responsibly
+
+{% include "web/tools/lighthouse/audits/implementation-heading.html" %}
+
+If geolocation permission was already granted to a page before Lighthouse's
+audit, Lighthouse cannot determine if the page requests the user's location
+on page load. Reset the permissions and run Lighthouse again.
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/geolocation-on-load.md]: "Reset the permission to the default behavior and run Lighthouse again."

Can we link to a doc on how to nuke permissions? On desktop, it's under the lock icon:

screen shot 2016-12-07 at 10 32 26 am

File: src/content/en/tools/lighthouse/audits/geolocation-on-load.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+Lighthouse collects the JavaScript that was executed on page load. If this
+code contains calls to `geolocation.getCurrentPosition()` or
+`geolocation.watchPosition()`, then the user's location was requested.
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/geolocation-on-load.md]: ....calls to `geolocation.getCurrentPosition()` or `geolocation.watchPosition()` and the Geolocation permission has not been granted yet, then the ....

File: src/content/en/tools/lighthouse/audits/http2.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+To pass this audit, serve each of those resources over HTTP/2.
+
+To learn how to enable HTTP/2 on your servers, see [Setting Up HTTP/2][setup].
+
+[setup]: https://dassur.ma/things/h2setup/
+
+{% include "web/tools/lighthouse/audits/implementation-heading.html" %}
+
+Lighthouse gathers all of the resources that are from the same host as the
+page, and then uses the Chrome Debugger Protocol to check the HTTP protocol
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/http2.md]: I think you can remove the "uses the Chrome Debugger Protocol" detail.

File: src/content/en/tools/lighthouse/audits/notifications-on-load.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+{% include "web/tools/lighthouse/audits/implementation-heading.html" %}
+
+If notification permissions was already granted or denied to a page before
+Lighthouse's audit, Lighthouse cannot determine if the page requests
+notification permissions on page load. Reset the permissions and run
+Lighthouse again.
+
+Lighthouse collects the JavaScript that was executed on page load. If this
+code contains calls to `notification.requestPermission()`, then
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/notifications-on-load.md]: also if the permission was not already granted.

File: src/content/en/tools/lighthouse/audits/old-flexbox.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+description: Reference documentation for the "Site Does Not Use The Old CSS Flexbox" Lighthouse audit.
+
+{# wf_updated_on: 2016-12-05 #}
+{# wf_published_on: 2016-12-05 #}
+
+# Site Does Not Use The Old CSS Flexbox  {: .page-title }
+
+## Why the audit is important {: #why }
+
+The old, 2009 specification for Flexbox is deprecated and is 2.3 times slower
```
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/old-flexbox.md]: 2.3 times -> 2.3x

Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/old-flexbox.md]: just a suggestion

File: src/content/en/tools/lighthouse/_toc.yaml (hunk 1 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/_toc.yaml]: The current title matches the title in the audit source. If you want to change the audit source, I'll change this, else I want these doc titles to always match the audit source for SEO

Comment (kaycebasques) [inline src/content/en/tools/lighthouse/_toc.yaml]: Ditto. I wrote the doc title to match the audit title

File: src/content/en/tools/lighthouse/audits/blocking-links.md (hunk 4 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/blocking-links.md (hunk 5 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/_toc.yaml (hunk 1 above)
Comment (ebidel) [inline src/content/en/tools/lighthouse/_toc.yaml]: SGTM

Comment (ebidel) [inline src/content/en/tools/lighthouse/_toc.yaml]: ack

File: src/content/en/tools/lighthouse/audits/blocking-links.md (hunk 2 above)
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: But we have two audits, one for script and one for links.

screen shot 2016-12-07 at 11 16 16 am

File: src/content/en/tools/lighthouse/audits/console-time.md (hunk 6 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/console-time.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/http2.md (hunk 9 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/http2.md]: I usually just leave this in there on the off-chance that somebody needs to know how it was extracted

File: src/content/en/tools/lighthouse/audits/geolocation-on-load.md (hunk 7 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/geolocation-on-load.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/geolocation-on-load.md (hunk 8 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/geolocation-on-load.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/notifications-on-load.md (hunk 10 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/notifications-on-load.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/old-flexbox.md (hunk 11 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/old-flexbox.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/http2.md (hunk 9 above)
Comment (ebidel) [inline src/content/en/tools/lighthouse/audits/http2.md]: It's really an implementation detail of LH though. The whole project uses the CDP to audit pages :)

Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/http2.md]: :heavy_check_mark:

File: src/content/en/tools/lighthouse/audits/blocking-links.md (hunk 2 above)
Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: Ah, I see. I might want to split these up in the future, but this is fine for now. I'll change the title.

Comment (kaycebasques) [inline src/content/en/tools/lighthouse/audits/blocking-links.md]: :heavy_check_mark:

**Comment by kaycebasques** [comment]
OK @ebidel tx for review, I think I handled all your changes

**Comment by PaulKinlan** [comment]
@ebidel @kaycebasques is this ready to land?

**Comment by kaycebasques** [comment]
I'll do a copyedit today and then it'll be ready

**Pull request merged**